### ISSUE TITLE
liked lists are never updated

### ISSUE THREAD

**Issue by antonsoroko** [elgatito/plugin.video.elementum#711]
## Current Behavior

https://github.com/elgatito/elementum/blob/8ff732c/library/trakt.go#L776
`RefreshTraktLists` will never update liked lists.
but even if we add `trakt.Likedlists()` there - then it will add content of liked lists to the library, which is probably is not what people expect.

## Possible Solution

i would like to try to fix this either by skipping `writeMovieStrm` for liked lists or by forcing update (in `UserlistShows`) when person open liked list (without background update).

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used: 0.1.69

* Environment name and version: kodi 19

* Operating System and version: android tv 9

* Link to Kodi debug log:

**Comment by SubEdict** [comment]
Same. 
This is why I'm using IMDB Helper -Trakt -Movies -Your like list (or something similar) --> Play in Elementum.
* Also, It's a little strange because it's looks like it do update in some point(weeks.. months..?) but not in the right order.
e.g [Latest Releases](https://trakt.tv/users/giladg/lists/latest-releases?sort=added,asc) will show in **Rank** order instead of **Recently Added**.

**Comment by elgatito** [comment]
@antonsoroko `Refresh...` is only about sync with Kodi library.

**Comment by elgatito** [comment]
@antonsoroko Not understanding the issue. When you like a list, it will show up in "My lists". There is a mix with your lists and liked lists.

**Comment by antonsoroko** [comment]
so initially i saw it like this:

https://github.com/elgatito/elementum/blob/2b049cc/api/trakt.go#L211
code passes `false` as `isUpdateNeeded`
so data returned from the cache https://github.com/elgatito/elementum/blob/2b049cc/trakt/shows.go#L541
and data expected to be updated in background via RefreshTrakt() https://github.com/elgatito/elementum/blob/8ff732c/library/trakt.go#L25 and via RefreshTraktLists() for lists in particular (which uses SyncShowsList() that finally updates the cache).

but RefreshTraktLists invoke only `trakt.Userlists()`, but never `trakt.Likedlists()` thus liked lists are never updated in background.

The only thing I missed was that cache has expiration and for lists it is small - `TraktShowsListExpire`  https://github.com/elgatito/elementum/blob/96c2f7586b71427fcea20dfdc1d807cf6269750c/cache/types.go#L97

So code path that I described above is for the library update - like you said.

**Comment by antonsoroko** [comment]
So I guess the only optimization that can be done here is to increase `TraktShowsListExpire` to `GeneralExpire` like for other cache and then use `updated_at` property of list to determine if we really need to update cache.
If you don't mind - I will try to do this (just for fun).
Thanks.

**Comment by elgatito** [comment]
@antonsoroko 
> but RefreshTraktLists invoke only trakt.Userlists(), but never trakt.Likedlists() thus liked lists are never updated in background.

Yes, they are not updated in the background, because we don't user it anywhere, except the user requested list of Lists.

**Comment by antonsoroko** [comment]
@elgatito про https://github.com/elgatito/elementum/pull/104#issuecomment-1946571620

>Probably we will need to cache it for longer time, and it list belongs to currently-authorized user - we can check activities to validate if lists were changed to refresh the list.

сейчас вроде уже проверяется активности, и можно увеличить время кэша для собственных листов.
это я про этот фикс https://github.com/elgatito/elementum/commit/24498e7d26584bee3b8419925b9cd3e3246bef2e#diff-b6af6ec5b6065d2f17923a8c2083a6a350075573458281a733f2ce41cfb58693L210

НО для чужих листов стало странно, мы обновляем или не обновляем их основываясь на "нашей" собственной активности про наши собственные листы.

в общем-то тут я описал правильный, как мне кажется, путь - проверять `updated_at` самого листа (каждого по отдельности).
или из ответа trakt.Userlists()/trakt.Likedlists() вырезать дату для листа (там есть нужные метаданные) или делать запрос на `/users/user_slug/lists/list_slug` чтобы получить только метаданные. ну и кешировать и проверять `updated_at`.

сделал proof of concept сначала в лоб (но нашёл косяк) https://github.com/elgatito/elementum/compare/master...antonsoroko:elementum:improve-lists-caching?expand=1, но там же описал возможное правильное решение (то что выше я описал):
https://github.com/antonsoroko/elementum/blob/c516fef5b7e23b4b32e2392a9957c31ac52a8df0/api/movies.go#L324-L327

**Comment by elgatito** [comment]
@antonsoroko Там же кеширование содержимого списка всего на 24 часа вроде. Стоит ли утруждаться ради обновления хужих списков чаще чем 24 часа? 
Там в принципе даже поставить можно 6 часов, того с головой должно быть.

**Comment by antonsoroko** [comment]
@elgatito там сейчас 1 минута же
https://github.com/elgatito/elementum/blob/9770024ff6dc7f1e6e4ebfb70bb334bb7c2f735f/cache/types.go#L75

**Comment by elgatito** [comment]
https://github.com/elgatito/elementum/commit/97d3d654d54a91c4ace7ec6a21f3df908071b7bc

Я добавил проверку даты изменения списка чтоб правильно кеш использовать.
Но Trakt кеширует эти данные, и даже после изменения списка он не показывает изменения в списке (при запросе на `/lists/%id`).

Пока не вижу что еще можно сделать.

**Comment by antonsoroko** [comment]
@elgatito 
Для /lists/ в документации написано, что оно только для публичных. И ещё там не нужна авторизация , соответственно вероятно на этой стоит "долгий" кэш с их стороны.

Технически можно сделать /lists только для official, 
а /users/%user/lists/%id для всех остальных (как было раньше). Тогда с кешем не будет проблем, вроде там данные сразу отдаются.

**Comment by elgatito** [comment]
С последними изменениями работает уже нормально у меня.
